internal/tmpl/tree_test.go

```diff
@@ -119,6 +119,50 @@ func TestTStructInTemplate(t *testing.T) {
 	testExec(t, funcs, tmpl, want, plain, nil)
 }
 
+func TestBasicPostprocess(t *testing.T) {
+	funcs := htmltmpl.FuncMap{
+		"golive_postprocess_context_var": func() string { return "$ppctx" },
+		"golive_postprocess_start":       func() []string { return []string{"pp", "qq"} },
+		"golive_postprocess_end":         func() []string { return []string{"xpp"} },
+		"pp": func() string {
+			return "["
+		},
+		"qq": func() string {
+			return "("
+		},
+		"xpp": func(x ...any) string {
+			if len(x) == 0 {
+				return "]"
+			}
+			if x[0] != "!" {
+				t.Fatalf("$ppctx not set correctly, got %v (%T)", x[0], x[0])
+			}
+			return x[1].(string) + "]"
+		},
+	}
+
+	tests := []struct {
+		tmpl  string
+		want  string
+		plain string
+	}{
+		{
+			tmpl:  `{{$ppctx := "!"}}a{{ pp }}123{{ xpp }}b{{qq}}456{{ xpp }}c`,
```

Comment:
theoretically `$ppctx` could be `any` right?  is it worth testing a non-`string`?

Reply:
theoretically, yes. but the code as written would be hard to screw up in a way such that it worked only for strings, so i'd like to punt on that for now.